Makefile.am

```diff
@@ -115,3 +115,11 @@ noinst_HEADERS += defaults.h
 
 indent:
 	tools/indent.py `find sharpd bgpd eigrpd include isisd lib nhrpd ospf6d ospfd pimd qpb ripd vtysh zebra -name '*.[ch]' | grep -v include/linux`
+
+if HAVE_GCOV
+check-coverage: check
+	@ if [ "x${COMMIT}" = "x" ]; then echo "COMMIT required"; exit 1; fi
+	@ if [ "x${TOKEN}" = "x" ]; then echo "TOKEN required"; exit 1; fi
+	@ find . -name '*.o' -exec gcov {} \;
+	curl -s https://codecov.io/bash | bash -s - -C ${COMMIT} -t ${TOKEN}
```

Comment:
There should probably be a confirmation prompt here informing the user they're about to pipe from the web into bash.

Reply:
Ok, makes sense, thanks for having a look.  I'll be running this in a script in automation so I'd like to have to some way of running this without interaction.  How about a force-check-coverage target with no prompting and add the prompt for check-coverage Makefile target?

Another option would be to keep the upload out of the Makefile (check-coverage just runs check and gcov) and put it into the docker image (without prompting).

What do you think?

Reply:
A force target sounds fine to me.